sqlmesh/lsp/hints.py

```diff
@@ -0,0 +1,118 @@
+"""Type hinting on SQLMesh models"""
+
+import typing as t
+
+from lsprotocol import types
+
+from sqlglot import exp
+from sqlglot.expressions import Expression
+from sqlglot.optimizer.normalize_identifiers import normalize_identifiers
+from sqlglot.optimizer.scope import build_scope, find_all_in_scope
+from sqlmesh.core.model.definition import SqlModel
+from sqlmesh.lsp.context import LSPContext, ModelTarget
+from sqlmesh.lsp.uri import URI
+
+
+def get_hints(
+    lsp_context: LSPContext,
+    document_uri: URI,
+    start_line: int,
+    end_line: int,
+) -> t.List[types.InlayHint]:
+    """
+    Get type hints for certain lines in a document
+
+    Args:
+        lint_context: The LSP context
+        document_uri: The URI of the document
+        start_line: the starting line to get hints for
+        end_line: the ending line to get hints for
+
+    Returns:
+        A list of hints to apply to the document
+    """
+    path = document_uri.to_path()
+    if path.suffix != ".sql":
+        return []
+
+    if path not in lsp_context.map:
+        return []
+
+    file_info = lsp_context.map[path]
+
+    # Process based on whether it's a model or standalone audit
+    if isinstance(file_info, ModelTarget):
+        # It's a model
+        model = lsp_context.context.get_model(
+            model_or_snapshot=file_info.names[0], raise_if_missing=False
+        )
+        if model is None or not isinstance(model, SqlModel):
+            return []
+
+        query = model.query
+        dialect = model.dialect
+        columns_to_types = model.columns_to_types or {}
+    else:
+        return []
+
+    return _get_type_hints_for_model_from_query(
+        query, dialect, columns_to_types, start_line, end_line
+    )
+
+
+def _get_type_hints_for_model_from_query(
+    query: Expression,
+    dialect: str,
+    columns_to_types: t.Dict[str, exp.DataType],
+    start_line: int,
+    end_line: int,
+) -> t.List[types.InlayHint]:
+    hints: t.List[types.InlayHint] = []
+    try:
+        query = normalize_identifiers(query.copy(), dialect=dialect)
+        root = build_scope(query)
+
+        if not root:
+            return []
+
+        for select in find_all_in_scope(root.expression, exp.Select):
+            for select_exp in select.expressions:
+                if not select_exp:
+                    continue
```

Comment:
I don't think this can ever be falsey.

Reply:
Good catch, I added this when I was debugging my stupid `,` vs `.` problem lol